Automattic/wp-calypso
# Implement pricing plans block

#### Proposed Changes

* Adds Gutenberg block to display plans information and provide quick links for upgrading


Gutenberg editor block screenshot:

<img width="641" alt="image" src="https://user-images.githubusercontent.com/112691742/202114095-1a23b37a-0833-4867-8bcb-489d190832ac.png">

Post view sscreenshot:

<img width="745" alt="image" src="https://user-images.githubusercontent.com/112691742/202114862-af67f78f-44ec-456c-89d1-608da0d8b3c2.png">


Settings pane screenshot:

<img width="713" alt="image" src="https://user-images.githubusercontent.com/112691742/202115501-9e711f1f-5cf9-4ceb-b59f-05cea0ade5eb.png">



#### Testing Instructions

* Run `yarn dev --sync` in `apps/happy-blocks` to sync code to your sandbox
* On your sandbox, edit `wp-content/themes/a8c/supportforums/editor/gutenberg-wpcom.php` and add to the function `supportforums_gutenberg_setup_frontend` before `return $settings;` line:
```
$settings['iso']['blocks']['allowBlocks'][] = 'happy-blocks/pricing-plans';
```
* Create a new topic, in the editor press `/` and search for "Upgrade" block.
* Focus the block, and in the block settings pane (gear icon) pre-select one of the available plans. Ensure the block updates to the selected plan in the editor, then pre-select a billing option (yearly/monthly) for the user.
* Submit the post, ensure the block loads in the Post view and shows the correct plan and the pre-selected billing option.
* Click on the "Learn more" link and ensure you're shown all the pricing plans options (_warning: this might not work when sandboxed_)
* Click on the Upgrade CTA button and ensure you're presented with the checkout process with the right plan pre-added to the cart. (_warning: this might not work when sandboxed_)


#### Pre-merge Checklist






Related to #
